File: 3143-longest-unequal-adjacent-groups-subsequence-i/longest-unequal-adjacent-groups-subsequence-i.go
func getLongestSubsequence(words []string, groups []int) []string {
    if len(words) == 0 {
        return []string{};
    }

    result := []string{words[0]};
    flag := groups[0];

    for i := 0; i < len(groups); i++ {
        if flag != groups[i] {
            result = append(result, words[i]);
            flag = groups[i];
        }
    }

    return result;
}

File: 3143-longest-unequal-adjacent-groups-subsequence-i/longest-unequal-adjacent-groups-subsequence-i.java
class Solution {
    public List<String> getLongestSubsequence(String[] words, int[] groups) {
        List<String> result = new ArrayList<>();
        result.add(words[0]);
        int flag = groups[0];

        for (int i = 0; i < groups.length; i++) {
            if (groups[i] != flag) {
                result.add(words[i]);
                flag = groups[i];
            }
        }

        return result;
    }
}